src/data/fintype/basic.lean

```diff
@@ -829,6 +829,12 @@ lemma univ_option (α : Type*) [fintype α] : (univ : finset (option α)) = inse
   fintype.card (option α) = fintype.card α + 1 :=
 (finset.card_cons _).trans $ congr_arg2 _ (card_map _) rfl
 
+/-- A type is a `fintype` if its successor (using `option`) is a `fintype`. -/
+noncomputable
+def fintype_of_equiv_option [fintype α] (f : option β ≃ α) : fintype β :=
+fintype.of_injective (embedding.coe_option.trans f.to_embedding) $
+injective.comp (equiv.to_embedding f).injective embedding.coe_option.injective
```

Comment:
I think this proof should be a 1-liner using two rewrites:
```
fintype (option X) \iff fintype X
(e : X \equiv Y) : fintype X \iff fintype Y
```

Reply:
I've split this into two lemmas `fintype_of_option` and `fintype_of_equiv` in #13086.